collated/main/A0141021H.md

```diff
@@ -0,0 +1,471 @@
+            indicateAttemptToExecuteIncorrectCommand();
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+        ReadOnlyTask taskToMark = lastShownList.get(targetIndex - 1);
+        try {
+            model.unMarkTask(taskToMark);
+            model.getStackOfMarkUndoneTask().push(taskToMark);
+            model.getStackOfMarkUndoneTaskTaskType().push(taskType);
+        } catch (TaskNotFoundException pnfe) {
+            return new CommandResult(String.format(RedoCommand.MESSAGE_FAIL));
+        }
+        return new CommandResult(String.format(RedoCommand.MESSAGE_SUCCESS));
+    }
+}
+```
+###### \java\seedu\whatnow\model\Model.java
+``` java
+    void changeLocation(Path destination, Config config) throws DataConversionException, IOException, TaskNotFoundException;
+	
+}
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    /** Raises an event to indicate the config has changed */
+    private void indicateConfigChanged(Path destination, Config config) {
+        raise(new ConfigChangedEvent(destination, config));
+    }
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    @Override
+    public synchronized void changeLocation(Path destination, Config config) throws DataConversionException, IOException, TaskNotFoundException {
+        indicateConfigChanged(destination, config);
+        indicateWhatNowChanged();
+    }
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    @Override
+    public synchronized void unMarkTask(ReadOnlyTask target) throws TaskNotFoundException {
+        whatNow.unMarkTask(target);
+        indicateWhatNowChanged();
+    }
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    @Override
+    public Stack<ReadOnlyTask> getStackOfMarkUndoneTask() {
+        return stackOfMarkUndone;
+    }
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    public Stack<String> getStackOfMarkUndoneTaskTaskType() {
+        return stackOfMarkUndoneTaskTypes;
+    }
+```
+###### \java\seedu\whatnow\model\ModelManager.java
+``` java
+    interface Expression {
+        boolean satisfies(ReadOnlyTask task);
+        String toString();
+    }
+
+    private class PredicateExpression implements Expression {
+
+        private final Qualifier qualifier;
+
+        PredicateExpression(Qualifier qualifier) {
+            this.qualifier = qualifier;
+        }
+
+        @Override
+        public boolean satisfies(ReadOnlyTask task) {
+            return qualifier.run(task);
+        }
+
+        @Override
+        public String toString() {
+            return qualifier.toString();
+        }
+    }
+
+    interface Qualifier {
+        boolean run(ReadOnlyTask task);
+        String toString();
+    }
+```
+###### \java\seedu\whatnow\model\task\UniqueTaskList.java
+``` java
+    /**
+     * 
+     * Mark the equivalent task from the list as uncompleted.
```

Comment:
Mark-**s**_ ...
please fix similar errors in the header comments so that they all follow the guidelines for header comments